model/task_history.go

```diff
@@ -779,12 +781,60 @@ func testHistoryV2Results(params *TestHistoryParameters) ([]task.Task, error) {
 	tasks = append(tasks, oldTasks...)
 
 	// to join the test results, merge test results for all the tasks
-	testQuery := db.Query(formTestsQuery(params))
-	out, err := task.MergeTestResultsBulk(tasks, &testQuery)
-	if err != nil {
-		return nil, errors.Wrap(err, "error merging test results")
+	if len(params.TestNames) > 0 {
+		testQuery := db.Query(formTestsQuery(params))
+		out, err := task.MergeTestResultsBulk(tasks, &testQuery)
+		if err != nil {
+			return nil, errors.Wrap(err, "error merging test results")
+		}
+		return out, nil
+	} else {
+		out := []task.Task{}
+		taskChan := make(chan task.Task, len(tasks))
+		for _, t := range tasks {
+			taskChan <- t
+		}
+		close(taskChan)
+		wg := sync.WaitGroup{}
+		wg.Add(numQueryThreads)
+		resultChan := make(chan task.Task)
+		catcher := grip.NewSimpleCatcher()
+		for i := 0; i < numQueryThreads; i++ {
+			go func() {
+				defer wg.Done()
+				for t := range taskChan {
+					err := t.MergeNewTestResults() //nolint
```

Comment:
I think the lint error is the colon in the assignment. could also put this in the if statement

Reply:
I need to redeclare this in each goroutine or each will share the same address for this variable. I renamed it to make it cleaner